032-MLOpsFromScratch/Student/03-BuildPipeline.md

```diff
@@ -0,0 +1,89 @@
+# Challenge 3 – Create a Build Pipeline in Azure DevOps
+
+[< Previous Challenge](./02-UnitTesting.md) - **[Home](../README.md)** - [Next Challenge >](./04-ReleasePipeline.md)
+
+## Introduction
+
+The Build pipeline demonstrates the automation of various stages/tasks involved
+in building an ML model and creating a container image on top of that model. The
+stages generally constitute creating a project environment, preparing the data,
+training the model, evaluating the model, registering/versioning the model, and
+scoring the model generally by containerizing the model.
+
+There are several ways to create a Build pipeline. The two most common and popular ways
+are:
+
+-   using a YAML file that represents the entire pipeline,
+
+-   using an empty job and adding tasks sequentially
+
+We believe that the latter approach is more comprehensive and intuitive, especially to
+get started on MLOps, so we recommend that route.  This will be the focus of this hack.
+
+We can setup Continuous Integration (CI) trigger for every Build pipeline. The
+CI pipeline gets triggered every time code is checked in. It publishes an
+updated Azure Machine Learning pipeline after building the code.
+
+## Description
```

Comment:
This challenge text feels like a Hands On Lab.... do you really want to hold their hands this much?

Reply:
There are so many data scientists who are completely unfamiliar with CICD process or its triggers so I tried to keep the introductions very descriptive. :)
But I understand your point. I will leave it next week's hack and update it based on feedback/observations if that is okay.